pkg/verify/configmap.go

```diff
@@ -8,18 +8,81 @@ import (
 
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/openpgp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
 
 	"github.com/openshift/cluster-version-operator/pkg/verify/store"
+	"github.com/openshift/cluster-version-operator/pkg/verify/store/configmap"
 	"github.com/openshift/cluster-version-operator/pkg/verify/store/parallel"
 	"github.com/openshift/cluster-version-operator/pkg/verify/store/sigstore"
+	"github.com/openshift/cluster-version-operator/pkg/verify/util"
 )
 
-// ReleaseAnnotationConfigMapVerifier is an annotation set on a config map in the
-// release payload to indicate that this config map controls signing for the payload.
-// Only the first config map within the payload should be used, regardless of whether
-// it has data. See NewFromConfigMapData for more.
-const ReleaseAnnotationConfigMapVerifier = "release.openshift.io/verification-config-map"
+const (
+	// ReleaseAnnotationConfigMapVerifier is an annotation set on a config map in the
+	// release payload to indicate that this config map controls signing for the payload.
+	// Only the first config map within the payload should be used, regardless of whether
+	// it has data. See NewFromConfigMapData for more.
+	ReleaseAnnotationConfigMapVerifier = "release.openshift.io/verification-config-map"
+
+	// verifierPublicKeyPrefix is the unique portion of the key used within a config map
+	// identifying data field containing one or more GPG public keys in ASCII form that
+	// must have signed the release image by digest.
+	verifierPublicKeyPrefix = "verifier-public-key-"
+
+	// storePrefix is the unique portion of the key used within a config map identifying
+	// data field containing a URL (scheme http://, or https://) location that contains
+	// signatures.
+	storePrefix = "store-"
+)
+
+// LoadConfigMapVerifierDataFromManifest decodes the manifest as a config map and checks it for the correct
+// verifier annotations. If found returns data as a map otherwise returns an empty map.
+func LoadConfigMapVerifierDataFromManifest(manifestBytes []byte) map[string]string {
```

Comment:
How will `oc` use this?  Today it [iterates over manifests][1], [like we do in the CVO][2].  Can we use `oc`'s `LoadConfigMapVerifierDataFromUpdate` here instead?  Although maybe call it `NewFromManifests` or some such?

[1]: https://github.com/openshift/oc/blob/49e8811ef52d58451e317cd616bcbca1865505f6/pkg/helpers/release/verify.go#L341-L343
[2]: https://github.com/openshift/cluster-version-operator/blob/a49fef5c66c6b0707c54fd93f84d2f51d3d28aca/pkg/cvo/cvo.go#L277-L308

Reply:
I removed this method as oc will send in manifests like CVO does now. As a result, I broke out logic to create the verifier from CVO method loadConfigMapVerifierDataFromUpdate into a new configmap method NewFromManifests since oc only needs a verifier without stores or a persister.